### PR TITLE
SALTO-4376: Netsuite - Update validateCredentials to return accountType and isProduction info

### PR DESCRIPTION
**Pull request by shoshana-houri**
Use `systemInformation` as returned from SuiteApp in order to define the `accountType` and `isProduction` values.

---

_Additional context for reviewer_

A new SuiteApp version returns envType info - which is one of the 4 values - `production`, `sandbox`, `beta` or `internal`.
We use these values to define the account type and whether it's a production account or not.

Notice that `accountType` and `isProduction` values are optional and are not returned in cases the account does not have SuiteApp credentials.

---
_Release Notes_: 

---
_User Notifications_:

### CODE REVIEW

File: packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts [hunk 1]
```diff
@@ -406,13 +406,18 @@ export default class SuiteAppClient {
...
       credentials,
       globalLimiter: new Bottleneck(),
       instanceLimiter: () => false,
     })
-    await client.sendRestletRequest('sysInfo')
+    const sysInfo = await client.getSystemInformation()
+
+    if (sysInfo === undefined) {
+      throw new Error('Failed getting SuiteApp system information')
+    }
```
[the remark this answers is not in the document]
Comment (dantal4) [reply packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:419]: I'm fine with #2, but notice that if `client.sendRestletRequest('sysInfo')` throws a credentials error (`InvalidSuiteAppCredentialsError`) we should throw that error as well.

[the remark this answers is not in the document]
Comment (shoshana-houri) [reply packages/netsuite-adapter/src/client/suiteapp_client/suiteapp_client.ts:419]: That's a good point, thanks!

File: packages/netsuite-adapter/src/client/client.ts [hunk 2]
```diff
@@ -97,19 +97,30 @@ export default class NetsuiteClient {
 
   @NetsuiteClient.logDecorator
   static async validateCredentials(credentials: Credentials): Promise<AccountInfo> {
-    if (isSuiteAppCredentials(credentials)) {
-      try {
-        await SuiteAppClient.validateCredentials(credentials)
-      } catch (e) {
-        e.message = `Salto SuiteApp Authentication failed. ${e.message}`
-        throw new CredentialError(e.message)
+    const systemInformation = await (async () => {
```
[the remark this answers is not in the document]
Comment (shoshana-houri) [reply packages/netsuite-adapter/src/client/client.ts:100]: lol I really doubted what's better, I will :)

**Review by dantal4 — APPROVED** [review]
💯

**Pull request merged**